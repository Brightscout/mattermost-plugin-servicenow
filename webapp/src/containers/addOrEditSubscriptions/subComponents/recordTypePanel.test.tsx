import React from 'react';

import {shallow, ShallowWrapper} from 'enzyme';

import RecordTypePanel from './recordTypePanel';

const mockOnContinue = jest.fn();
const mockOnBack = jest.fn();
const mockSetRecordType = jest.fn();
const mockSetResetRecordPanelStates = jest.fn();

const mockRecordType: RecordType = 'incident';

const recordTypePanelProps = {
    className: 'mockClassName',
    error: 'mockError',
    onBack: mockOnBack,
    onContinue: mockOnContinue,
    actionBtnDisabled: true,
    requiredFieldValidationErr: true,
    recordType: mockRecordType,
    setRecordType: mockSetRecordType,
    setResetRecordPanelStates: mockSetResetRecordPanelStates,
};

describe('Record Type Panel', () => {
    let component: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

    beforeEach(() => {
        component = shallow(<RecordTypePanel {...recordTypePanelProps}/>);
    });

    afterEach(() => {
        component.unmount();
    });

    it('Should render correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('Should apply the passed className prop', () => {
        expect(component.hasClass(recordTypePanelProps.className)).toBeTruthy();
    });

    it('Should render the record type panel body correctly', () => {
        expect(component.find('Dropdown')).toHaveLength(1);
        expect(component.find('ModalSubtitleAndError')).toHaveLength(1);
        expect(component.find('ModalFooter')).toHaveLength(1);
    });
});
