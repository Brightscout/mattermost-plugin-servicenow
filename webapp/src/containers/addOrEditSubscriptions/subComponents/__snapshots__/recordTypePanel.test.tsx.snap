// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Record Type Panel Should render correctly 1`] = `
<div
  className="modal__body wizard__secondary-panel mockClassName"
>
  <div
    className="padding-h-12 padding-v-20 wizard__body-container"
  >
    <Dropdown
      error={false}
      onChange={[Function]}
      options={
        Array [
          Object {
            "label": "Incident",
            "value": "incident",
          },
          Object {
            "label": "Problem",
            "value": "problem",
          },
          Object {
            "label": "Change Request",
            "value": "change_request",
          },
        ]
      }
      placeholder="Select Record Type"
      required={true}
      value="incident"
    />
    <ModalSubtitleAndError
      error="mockError"
    />
  </div>
  <ModalFooter
    cancelBtnText="Back"
    cancelDisabled={true}
    confirmBtnText="Continue"
    confirmDisabled={true}
    onConfirm={[Function]}
    onHide={[MockFunction]}
  />
</div>
`;
